qiskit/qpy/formats.py

```diff
@@ -139,6 +139,11 @@
 PAULI_EVOLUTION_DEF_PACK = "!Q?1cQQ"
 PAULI_EVOLUTION_DEF_SIZE = struct.calcsize(PAULI_EVOLUTION_DEF_PACK)
 
+# Modifier
+MODIFIER_DEF = namedtuple("MODIFIER_DEF", ["type", "num_ctrl_qubits", "ctrl_state", "power"])
+MODIFIER_DEF_PACK = "!1cIId"
+MODIFIER_DEF_SIZE = struct.calcsize(MODIFIER_DEF_PACK)
```

Comment:
I think this requires bumping the QPY version to 11 as this is a new struct to represent the modifiers. If someone tried to load a file using this with 0.45 which uses the same qpy version this would result in an error when the loader encounters the annotated operation (actually the new type key would trigger this too).

Reply:
You'll need to change that condition, also bump https://github.com/Qiskit/qiskit/blob/main/qiskit/qpy/common.py#L23 to 11, and also update the format documentation to describe what's new in version 11. I have another PR #11646 that actually is adding version 11 already, you can use this as a model. The two will merge conflict but we can handle that as there will only be version 11 in this release (there is only one format version for each qiskit release).

#11644 will have implications for this, as we'll need to error if someone uses an annotated operation and the `version < 11` (this was something I didn't have to worry about in #11646)